titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/PodToTaskMapper.java

```diff
@@ -241,11 +246,43 @@ private Either<TaskStatus, String> handlePodRunning() {
         if (isBefore(Started, taskState)) {
             return unexpected("pod state (Running) not consistent with the task state");
         }
+
+        long now = titusRuntime.getClock().wallTime();
+        if ("NodeLost".equals(podWrapper.getReason())) {
+            // If task is still in the Accepted state, move it first, as we know it was scheduled.
+            if (taskState == TaskState.Accepted) {
+                return Either.ofValue(TaskStatus.newBuilder()
+                        .withState(Launched)
```

Comment:
Wait, `Launched` ?
How bad would it be to have a new state like `Disconnected` ?

Reply:
Hmm. I don't feel good about "lying" that we got Launched when we never heard that from the executor.

Reply:
I think the launched state is the best one, as the task is assigned to a scheduler which means it moved out of Accepted state. We do not know if the task moved to StartInitiated or Running phase.